src/API/Google/AdsAssetGroup.php

```diff
@@ -82,17 +86,56 @@ public function __construct( GoogleAdsClient $client, AdsAssetGroupAsset $asset_
 		$this->asset_group_asset = $asset_group_asset;
 	}
 
+	/**
+	 * Create an asset group.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int $campaign_id
+	 *
+	 * @return id The asset group id.
+	 * @throws ExceptionWithResponseData ExceptionWithResponseData When an ApiException is caught.
+	 */
+	public function create_asset_group( int $campaign_id ): int {
+		try {
+			$campaign_resource_name = ResourceNames::forCampaign( $this->options->get_ads_id(), $campaign_id );
+			$current_date_time      = ( new DateTime( 'now', wp_timezone() ) )->format( 'Y-m-d H:i:s' );
+			$name                   = sprintf(
+				/* translators: %s: current date time. */
+				__( 'PMax %s', 'google-listings-and-ads' ),
+				$current_date_time
+			);
+
+			$operations = $this->create_operations( $campaign_resource_name, $name );
+			return $this->mutate( $operations );
+
+		} catch ( ApiException $e ) {
+			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
+
+			$errors = $this->get_api_exception_errors( $e );
+			throw new ExceptionWithResponseData(
+				/* translators: %s Error message */
+				sprintf( __( 'Error creating asset group: %s', 'google-listings-and-ads' ), reset( $errors ) ),
+				$this->map_grpc_code_to_http_status_code( $e ),
+				null,
+				[
+					'errors' => $errors,
+				]
+			);
+		}
+	}
+
 	/**
 	 * Returns a set of operations to create an asset group.
 	 *
 	 * @param string $campaign_resource_name
-	 * @param string $campaign_name
+	 * @param string $name The asset group name.
 	 * @return array
 	 */
-	public function create_operations( string $campaign_resource_name, string $campaign_name ): array {
+	public function create_operations( string $campaign_resource_name, string $name ): array {
```

Comment:
Same as in https://github.com/woocommerce/google-listings-and-ads/pull/1851/files#r1071013296

💅  I'd personally would change `$name` to `$asset_group_name` since $name is quite generic.